### PR TITLE
various code smell fixes from sonarqube.

### PR DESCRIPTION
**Pull request by stuart-bass-bjss**

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Infrastructure/ValueObject.cs [hunk 1]
```diff
@@ -1,5 +1,7 @@
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.InteropServices.ComTypes;
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.Infrastructure/ValueObject.cs]: Do we need this?

File: tests/NHSD.BuyingCatalogue.Testing.Data/ConnectionAwaiter.cs [hunk 2]
```diff
@@ -6,6 +6,21 @@ namespace NHSD.BuyingCatalogue.Testing.Data
 {
     internal static class ConnectionAwaiter
     {
+        public class ConnectionAwaiterTimeoutException : Exception
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Testing.Data/ConnectionAwaiter.cs]: As discussed not sure we need this?

File: tests/NHSD.BuyingCatalogue.Testing.Data/ConnectionAwaiter.cs [hunk 3]
```diff
@@ -26,7 +41,7 @@ public static async Task AwaitConnectionAsync(string connectionString, int timeo
                 }
             }
 
-            throw new Exception($"Could not connect to database instance after trying for {timeout} seconds");
+            throw new ConnectionAwaiterTimeoutException($"Could not connect to database instance after trying for {timeout} seconds");
```
Comment (sam-roberts-bjss) [inline tests/NHSD.BuyingCatalogue.Testing.Data/ConnectionAwaiter.cs]: Could use a timeout exception rather than a custom one

**Comment by sam-roberts-bjss** [comment]
Also please be aware there a merge conflicts to fix

File: src/NHSD.BuyingCatalogue.Infrastructure/ValueObject.cs [hunk 4]
```diff
@@ -4,18 +4,18 @@
 namespace NHSD.BuyingCatalogue.Infrastructure
 {
     /// <summary>
-    /// Represents a building block of a domain model. It should be a small concept representing the value of an entity.
+    ///     Represents a building block of a domain model. It should be a small concept representing the value of an entity.
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.Infrastructure/ValueObject.cs]: Is the indent a part of the Resharper code clean up?

File: src/NHSD.BuyingCatalogue.SolutionLists.Application/Domain/SolutionList.cs [hunk 5]
```diff
@@ -1,32 +1,30 @@
...
+                .ToList();
         }
 
-        private IEnumerable<SolutionListItem> CapabilityFilter(ISet<Guid> capabilityIdList, IEnumerable<SolutionListItem> solutions)
-        {
-            return capabilityIdList.Any() ?
-                solutions.Where(solution => capabilityIdList.Intersect(solution.Capabilities.Select(capability => capability.Id)).Count() == capabilityIdList.Count()) :
-                solutions;
-        }
+        public List<SolutionListItem> Solutions { get; }
```
Comment (sam-roberts-bjss) [inline src/NHSD.BuyingCatalogue.SolutionLists.Application/Domain/SolutionList.cs]: Could we move this property back to the top? Initially I tried to be consistent with properties at the top of a class

File: src/NHSD.BuyingCatalogue.Infrastructure/ValueObject.cs (hunk 4 above)
Comment (stuart-bass-bjss) [reply src/NHSD.BuyingCatalogue.Infrastructure/ValueObject.cs]: Yes

File: src/NHSD.BuyingCatalogue.SolutionLists.Application/Domain/SolutionList.cs (hunk 5 above)
Comment (stuart-bass-bjss) [reply src/NHSD.BuyingCatalogue.SolutionLists.Application/Domain/SolutionList.cs]: reordered & applied change to dotsettings file.

**Pull request merged**